inference-engine/src/transformations/src/transformations/op_conversions/convert_broadcast3.cpp

```diff
@@ -11,50 +11,101 @@
+            input = std::make_shared<opset1::Broadcast>(input, target_shape_input, op::AutoBroadcastType::NUMPY);
         } else if (broadcast_type == op::BroadcastType::PDPD) {
-            last_node = std::make_shared<ngraph::opset1::Broadcast>(input, target_shape, op::AutoBroadcastType::PDPD);
-            ngraph::copy_runtime_info(broadcast, last_node);
+            input = std::make_shared<opset1::Broadcast>(input, target_shape_input, op::AutoBroadcastType::PDPD);
         } else if (broadcast_type == op::BroadcastType::NONE) {
-            last_node = std::make_shared<ngraph::opset1::Broadcast>(input, target_shape, broadcast->input_value(2), op::AutoBroadcastType::NONE);
-            ngraph::copy_runtime_info(broadcast, last_node);
+            input = std::make_shared<opset1::Broadcast>(input, target_shape_input, broadcast->input_value(2), op::AutoBroadcastType::NONE);
         } else if (broadcast_type == op::BroadcastType::BIDIRECTIONAL) {
-            auto constant_one = std::make_shared<ngraph::opset1::Constant>(input.get_element_type(), Shape({1}), std::vector<int>{1});
-            auto broadcast_ones = std::make_shared<ngraph::opset1::Broadcast>(constant_one, target_shape, op::AutoBroadcastType::NUMPY);
-            if (input.get_element_type() == element::boolean) {
-                last_node = std::make_shared<ngraph::opset1::LogicalOr>(input, broadcast_ones);
+            if (auto const_target_shape = std::dynamic_pointer_cast<opset1::Constant>(target_shape_input.get_node_shared_ptr())) {
+                const auto & input_shape = input.get_partial_shape();
+                const auto & target_shape = const_target_shape->cast_vector<size_t>();
+                std::vector<size_t> aligned_target_shape{target_shape};
+                if (make_compatible_shape(input_shape, aligned_target_shape)) {
+                    input = std::make_shared<opset1::Broadcast>(input,
+                            opset1::Constant::create(element::i64, Shape({aligned_target_shape.size()}), aligned_target_shape));
+                } else {
+                    input = std::make_shared<opset1::Multiply>(input,
+                            opset1::Constant::create(input_element_type, target_shape, {1}));
+                }
             } else {
-                last_node = std::make_shared<ngraph::opset1::Multiply>(input, broadcast_ones);
+                auto constant_one = opset1::Constant::create(input_element_type, {1}, {1});
+                auto broadcast_ones = std::make_shared<opset1::Broadcast>(constant_one, target_shape_input);
+                if (input_element_type == element::boolean) {
+                    input = std::make_shared<ngraph::opset1::LogicalOr>(input, broadcast_ones);
+                } else {
+                    input = std::make_shared<ngraph::opset1::Multiply>(input, broadcast_ones);
+                }
+                copy_runtime_info(broadcast, broadcast_ones);
```

Comment:
Can we add unit tests for this case (when `target_shape_input` is not Constant)?

Reply:
Tests were added.